internal/app/referenceclient/impl.go

```diff
@@ -184,14 +184,11 @@ func doUnary[ReqT, RespT any, Req pointerMessage[ReqT]](
 		trailers = internal.ConvertToProtoHeader(connectErr.Meta())
 		protoErr = internal.ConvertConnectToProtoError(connectErr)
 	} else {
-		// If the call was successful, get the headers and trailers
+		// If the call was successful, get the returned payloads
+		// and the headers and trailers
+		payloads = append(payloads, getPayload(resp.Msg))
```

Comment:
This was minor cleanup. As it was, it would have return a slice of zero payloads, even though there was a single response message (so it should have been a slice of one).

Adding nil is okay here. In fact, the client-stream case looked exactly like this before and worked just fine. So I just updated this to look exactly like the code in the client-stream handler.